gpii/node_modules/flowManager/src/Capture.js

```diff
@@ -0,0 +1,283 @@
+/**
+ * GPII snapshot Handler
+ *
+ * Copyright 2014 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/gpii/universal/LICENSE.txt
+ */
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.defaults("gpii.flowManager.capture", {
+    gradeNames: ["fluid.component"],
+    events: {
+        onPrepareCapture: null,
+        onPrepareSolutions: null,
+        onSolutions: null,
+        onReadyToCapture: {
+            events: {
+                prepareCapture: "onPrepareCapture",
+                prepareSolutions: "onPrepareSolutions",
+                solutions: "onSolutions"
+            },
+            args: [
+                "{arguments}.solutions.0",
+                "{arguments}.prepareCapture.0"
+            ]
+        },
+        onError: null,
+
+        // Newer versions below using fireTransformEvent
+        onSolutionsForCurrentDevice: null,
+        onCaptureSettingsForCurrentDevice: null
+    },
+    listeners: {
+        onSolutionsForCurrentDevice: [
+            {
+                funcName: "gpii.lifecycleManager.getDeviceContextPromise",
+                args: ["{flowManager}.deviceReporter"]
+            },
+            {
+                funcName: "gpii.flowManager.getSolutionsPromise",
+                args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"]
+            },
+            {
+                funcName: "fluid.toPromise",
+                args: ["{arguments}.0.solutionsRegistryEntries"]
+            }
+        ],
+        onCaptureSettingsForCurrentDevice: [
+            {
+                funcName: "gpii.lifecycleManager.getDeviceContextPromise",
+                args: ["{flowManager}.deviceReporter"]
+            },
+            {
+                funcName: "gpii.flowManager.getSolutionsPromise",
+                args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0"]
+            },
+            {
+                funcName: "fluid.toPromise",
+                args: ["{arguments}.0.solutionsRegistryEntries"]
+            },
+            {
+                func: "{that}.captureSystemSettingsPromise",
+                args: ["{arguments}.0", "{arguments}.1"]
+            }
+        ],
+        "onPrepareCapture.deviceContext": {
+            funcName: "gpii.lifecycleManager.getDeviceContext",
+            args: ["{flowManager}.deviceReporter", "{that}.events.onPrepareSolutions"]
+        },
+        onPrepareSolutions: {
```

Comment:
It seems you've decided to use `fluid.promise.fireTransformEvent()` for the implementation. If so, can you clean up this old version from line 77-88 as well as these events? Thanks.

Reply:
Removed older event based version.